ert3/evaluator/_evaluator.py

```diff
@@ -1,57 +1,136 @@
+        return self._fail_message
+
+    def getPhaseName(self) -> str:
+        return self._phase_name
+
+    def currentPhase(self) -> int:
+        return self._phase
+
+    def phaseCount(self) -> int:
+        return self._phase_count
+
+    def setPhase(
+        self, phase: int, phase_name: str, indeterminate: Optional[bool] = None
+    ) -> None:
+        self._phase_name = phase_name
+        if not 0 <= phase <= self._phase_count:
+            raise ValueError(
+                "Phase must be an integer between (inclusive) 0 and {self._phase_count}"
+            )
+
+        if phase == 0:
+            self._job_start_time = int(time.time())
+
+        if phase == self._phase_count:
+            self._job_stop_time = int(time.time())
+
+        self._phase = phase
+
+    def isIndeterminate(self) -> bool:
+        return False
+
+    def setIndeterminate(self, indeterminate: Union[bool, None]) -> None:
+        raise NotImplementedError("This function is not in use in ert3")
+
+
 def _run(
     ensemble_evaluator: EnsembleEvaluator,
+    run_model: DummyBaseRunModel,
 ) -> Dict[int, Dict[str, ert.data.RecordTransmitter]]:
     result: Dict[int, Dict[str, ert.data.RecordTransmitter]] = {}
     with ensemble_evaluator.run() as monitor:
-        realization_ids = set()
-        realizations_completed = set()
+        run_model.setPhase(0, "Running simulations...", indeterminate=False)
         for event in monitor.track():
             if isinstance(event.data, dict) and event.data.get("status") in [
                 ENSEMBLE_STATE_STOPPED,
                 ENSEMBLE_STATE_FAILED,
             ]:
                 monitor.signal_done()
                 if event.data.get("status") == ENSEMBLE_STATE_FAILED:
+                    run_model._failed = True
+                    run_model._fail_message = "Ensemble evaluation failed"
                     raise RuntimeError("Ensemble evaluation failed")
             if event["type"] == EVTYPE_EE_TERMINATED and isinstance(event.data, bytes):
+                run_model.setPhase(1, "Simulations completed.")
                 result = pickle.loads(event.data)
-            if isinstance(event.data, dict) and "reals" in event.data:
-                for real_id in event.data["reals"]:
-                    realization_ids.add(real_id)
-                    real_status = event.data["reals"][real_id].get("status")
-                    if (
-                        real_status == REALIZATION_STATE_FINISHED
-                        and real_id not in realizations_completed
-                    ):
-                        realizations_completed.add(real_id)
-                        print(
-                            f"Realization {real_id} completed successfully"
-                            f" ({len(realizations_completed)}/{len(realization_ids)})"
-                        )
 
     return result
 
 
 def evaluate(
-    ensemble: Ensemble,
+    ensemble: Ensemble, cli: bool = True
```

Comment:
What is the `cli` flag? Is it ever used?

Reply:
It is used, but it might be set as default and instead having a `gui` there?

Reply:
I'm not sure what you mean?

Reply:
If you check line 115, EvaluatorTracker is not created by default but only when `cli==True`. It could be created by default, but do something special when we will go after gui.